test/Driver/mms3.jl

```diff
@@ -0,0 +1,197 @@
+        V_g(t, x1, x2, x3, Val(3)),
+        W_g(t, x1, x2, x3, Val(3)),
+    )
+    state.ρe = E_g(t, x1, x2, x3, Val(3))
+end
+
+function mms3_source!(
+    bl,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+)
+    x1, x2, x3 = aux.coord
+    source.ρ = Sρ_g(t, x1, x2, x3, Val(3))
+    source.ρu = SVector(
+        SU_g(t, x1, x2, x3, Val(3)),
+        SV_g(t, x1, x2, x3, Val(3)),
+        SW_g(t, x1, x2, x3, Val(3)),
+    )
+    source.ρe = SE_g(t, x1, x2, x3, Val(3))
+end
+
+function main()
+    CLIMA.init()
+
+    FT = Float64
+
+    # DG polynomial order
+    N = 4
+
+    t0 = FT(0)
+    timeend = FT(1)
+
+    ode_dt = 3.125e-4
+    nsteps = ceil(Int64, timeend / ode_dt)
+    ode_dt = timeend / nsteps
+
+    ode_solver =
+        CLIMA.ExplicitSolverType(solver_method = LSRK54CarpenterKennedy)
+
+    expected_result = FT(9.176181458773599e-5)
+
+    model = AtmosModel{FT}(
+        AtmosLESConfigType;
+        orientation = NoOrientation(),
+        ref_state = NoReferenceState(),
+        turbulence = ConstantViscosityWithDivergence(FT(μ_exact)),
+        moisture = MMSDryModel(),
+        source = mms3_source!,
+        boundarycondition = InitStateBC(),
+        init_state = mms3_init_state!,
+        param_set = ParameterSet{FT}(),
+    )
+
+    brickrange = (
+        range(FT(0); length = 17, stop = 1),
+        range(FT(0); length = 17, stop = 1),
+        range(FT(0); length = 17, stop = 1),
+    )
+    topl = BrickTopology(
+        MPI.COMM_WORLD,
+        brickrange,
+        periodicity = (false, false, false),
+    )
+    warpfun =
+        (x1, x2, x3) -> begin
+            (
+                x1 + (x1 - 1 / 2) * cos(2 * π * x2 * x3) / 4,
+                x2 + exp(sin(2π * (x1 * x2 + x3))) / 20,
+                x3 + x1 / 4 + x2^2 / 2 + sin(x1 * x2 * x3),
+            )
+        end
+    grid = DiscontinuousSpectralElementGrid(
+        topl,
+        FloatType = FT,
+        DeviceArray = CLIMA.array_type(),
+        polynomialorder = N,
+        meshwarp = warpfun,
+    )
+    driver_config = CLIMA.DriverConfiguration(
+        AtmosLESConfigType(),
+        "MMS3",
+        N,
+        FT,
+        CLIMA.array_type(),
+        ode_solver,
+        model,
+        MPI.COMM_WORLD,
+        grid,
+        Rusanov(),
+        CentralNumericalFluxDiffusive(),
+        CentralNumericalFluxGradient(),
+        CLIMA.AtmosLESSpecificInfo(),
+    )
+
+    solver_config = CLIMA.SolverConfiguration(
+        t0,
+        timeend,
+        driver_config,
+        ode_solver_type = ode_solver,
+        ode_dt = ode_dt,
+    )
+    dg = DGModel(
+        model,
+        grid,
+        Rusanov(),
+        CentralNumericalFluxDiffusive(),
+        CentralNumericalFluxGradient(),
+    )
+
+    CLIMA.invoke!(solver_config)
+
+    Qe = init_ode_state(dg, timeend)
+    result = euclidean_distance(solver_config.Q, Qe)
+    @test result ≈ expected_result
+end
+
+main()
```

Comment:
This test is identical to the existing MMS test right (for a specific mesh resolution) ?

Reply:
Yes, and also 64 bit only. I'm not terribly happy about replicating this. What we probably need is a generalized MMS test that can be triggered with different options by different tests.

Reply:
Opened #910 to track this.